### PR TITLE
fix(chart): fix chart row style being incorrect when using Axis::title

### PR DESCRIPTION
**Pull request by aatukaj**
Fixes: #379

old behavior
image

new fixed
image

### CODE REVIEW

File: src/widgets/chart.rs [hunk 1]
```diff
@@ -560,7 +560,7 @@ impl<'a> Widget for Chart<'a> {
 
         if let Some((x, y)) = layout.title_x {
             let title = self.x_axis.title.unwrap();
-            let width = graph_area.right().saturating_sub(x);
+            let width = title.width() as u16;
```
Comment (joshka) [inline src/widgets/chart.rs:563]: How does this act when title.width is > buffer width? Can you please add a test to ensure that this works fine without panicking?

Comment (aatukaj) [reply src/widgets/chart.rs:563]: The title is hidden when it's too long. It's checked [here](https://github.com/ratatui-org/ratatui/blob/e098731d6c1a68a0319d544301ac91cf2d05ccb2/src/widgets/chart.rs#L302C13-L302C13).

**Comment by aatukaj** [comment]
I'm not really sure on how to go about fixing that commit lint :(

File: src/widgets/chart.rs (hunk 1 above)
Comment (joshka) [reply src/widgets/chart.rs:563]: Thanks for implementing this. The code that sets up the invariant is quite a long way (160 lines prior) to the code that relies on it.

**Comment by joshka** [comment]
> I'm not really sure on how to go about fixing that commit lint :(

Mainly this would be fixed by changing the word "added" to "add"
In this case, I'll squash the two commits during the merge, and add some context to the commit message body, as the additional test really is part of the main change and doesn't really need a separate commit.

**Review by joshka — APPROVED** [review]

**Comment by joshka** [comment]
Actually - change of plans, double checking this locally, the conventional commit lint prevented the formatting check to run, and there's an extraneous newline that the formatter corrects. Can you please squash the two commits using git rebase and then run `cargo +nightly fmt --all` and force push the result to your github branch (`git push origin main --force`)?

Do you think it would be worth adding another unit test that also checks that the data style is not overwritten by the title / axis style?

Here's a suggestion for an improved commit message. I'm not sure if I've correctly noted that it's the dataset style not the title style here.

```text
fix(chart): use graph style for top line

A bug in the rendering caused the top line of the chart to be rendered
using the style of the Y Axis, instead of the dataset style. This is
fixed by only setting the style for the width of the text, and not the
entire row.

Fixes: https://github.com/ratatui-org/ratatui/issues/379
```

**Review by joshka — CHANGES_REQUESTED** [review]
Actually - change of plans, double checking this locally, the conventional commit lint prevented the formatting check to run, and there's an extraneous newline that the formatter corrects. Can you please squash the two commits using git rebase and then run `cargo +nightly fmt --all` and force push the result to your github branch (`git push origin main --force`)?

Do you think it would be worth adding another unit test that also checks that the data style is not overwritten by the title / axis style?

Here's a suggestion for an improved commit message. I'm not sure if I've correctly noted that it's the dataset style not the title style here.

```text
fix(chart): use graph style for top line

A bug in the rendering caused the top line of the chart to be rendered
using the style of the Y Axis, instead of the dataset style. This is
fixed by only setting the style for the width of the text, and not the
entire row.

Fixes: https://github.com/ratatui-org/ratatui/issues/379
```

**Comment by aatukaj** [comment]
Added the test. Hopefully I did the rebase correctly. Thanks for the commit suggestion.

**Review by joshka — APPROVED** [review]

**Comment by joshka** [comment]
Thanks for adding the test. LGTM. One last thing (unless the other maintainers have comments on the PR):

We use commit signature verification, which will block commits from being merged unless they are signed. To set up your machine to sign commits, see [managing commit signature verification](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) in GitHub docs.

Also not a big deal, the commit message doesn't really need the following lines as writing tests can be generally considered as part of a feature:

```
test(chart): add test for too big titles
test(chart): add test for correct top line styling
```

**Review by orhun — APPROVED** [review]

**Comment by aatukaj** [comment]
Should be all good now.
Thanks for all the help!

**Pull request merged**